openedx/core/lib/safe_lxml/etree.py

```diff
@@ -16,7 +16,7 @@
 
 from lxml.etree import XMLParser as _XMLParser
 from lxml.etree import *  # lint-amnesty, pylint: disable=redefined-builtin
-from lxml.etree import _Element, _ElementTree
+from lxml.etree import _Comment, _Element, _ElementTree, _Entity, _ProcessingInstruction
```

Comment:
Let's see if I understand this...

- safe_lxml monkeypatches lxml at startup in order to get safer defaults
- We have to import everything from lxml.etree into this module in order to re-expose it, but the "private" members don't get imported by default
- The newer python-saml3 makes reference to some additional private module members (https://github.com/SAML-Toolkits/python3-saml/compare/v1.9.0...v1.15.0#diff-969a1a2b18d88cfbe94894951b2eab2de7494a907905878f0a9d32a0c19836ffR67) as part of its own lxml-defusing wrapper, so we have to include them in this list as well

Does that sound about right?

Reply:
Yup, that is pretty much what's going on here. I can make a comment explaining all of this.